pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -1431,6 +1431,21 @@ type NodeGroupBase struct {
 	// Enable EC2 detailed monitoring
 	// +optional
 	EnableDetailedMonitoring *bool `json:"enableDetailedMonitoring,omitempty"`
+
+	// CapacityReservation defines reservation policy for a nodegroup
+	CapacityReservation *CapacityReservation `json:"capacityReservation,omitempty"`
+}
+
+// CapacityReservation defines reservation policy for a nodegroup
+type CapacityReservation struct {
+	CapacityReservationPreference *string                    `json:"capacityReservationPreference,omitempty"`
+	CapacityReservationTarget     *CapacityReservationTarget `json:"capacityReservationTarget,omitempty"`
+}
+
+// CapacityReservationTarget defines reservation target details
+type CapacityReservationTarget struct {
+	CapacityReservationID               *string `json:"capacityReservationId,omitempty"`
```

Comment:
```suggestion
	CapacityReservationID               *string `json:"capacityReservationID,omitempty"`
```

shouldn't this be uppercase `ID`?

Reply:
Sure. :D Adjust the tests too :)